netlify/next-runtime
# fix: use process.env.URL when deploying to production

### Summary

* Upgrades the version to `next-auth` to the latest
* Updates the logic for creating the default `NEXTAUTH_URL`. 
  * In a production context, the unique hash that gets pre-pended to the URL leads to issues because `next-auth` uses that URL as the basis for updating it's provider information. For instance - the callback URL used for OAuth applications (since it expects a stable value). This change makes the URL stable by using the ['URL' rather than the `DEPLOY_PRIME_URL`](https://docs.netlify.com/configure-builds/environment-variables/#deploy-urls-and-metadata) when it detects that the [`CONTEXT`](https://docs.netlify.com/configure-builds/environment-variables/#build-metadata) is set to 'production'

### Test plan

1. Visit the Deploy Preview - https://deploy-preview-1680--netlify-plugin-nextjs-next-auth-demo.netlify.app/
2. Visit the '/admin' and '/me' pages.

* Visiting those pages should not result in a server error
* If you inspect the 'sign in' button shown as part of the default demo when visiting `/admin` or `/me`, the URL that it goes to should be the URL of the site (e.g: `https://deploy-preview-1680--netlify-plugin-nextjs-next-auth-demo.netlify.app/api/callback....`)

1. Visiting this test site https://upgrade-next-auth--ep-next-auth.netlify.app/
2. Visit the `/admin` and `/me` pages

* Visiting those pages should not result in a server error
* If you inspect the 'sign in button' on `/admin` or `/me`, the URL that it goes to should start with `https://ep-next-auth.netlify.app/` not `https://upgrade-next-auth--ep-next-auth.netlify.app`
* If you sign in (there's a test OAuth app backing that site), you should be able to visit the `/me` endpoint, but the `/admin` page should still be inaccessible since `userRole: admin` is not set on the returned token that makes that page accessible.
